fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/type/OperatorUtil.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.persistence.jdbc.util.type;
+
+import static com.ibm.fhir.persistence.jdbc.JDBCConstants.EQ;
+import static com.ibm.fhir.persistence.jdbc.JDBCConstants.GT;
+import static com.ibm.fhir.persistence.jdbc.JDBCConstants.GTE;
+import static com.ibm.fhir.persistence.jdbc.JDBCConstants.LIKE;
+import static com.ibm.fhir.persistence.jdbc.JDBCConstants.LT;
+import static com.ibm.fhir.persistence.jdbc.JDBCConstants.LTE;
+import static com.ibm.fhir.persistence.jdbc.JDBCConstants.NE;
+import static com.ibm.fhir.persistence.jdbc.JDBCConstants.modifierOperatorMap;
+
+import com.ibm.fhir.database.utils.query.Operator;
+import com.ibm.fhir.search.SearchConstants.Modifier;
+import com.ibm.fhir.search.SearchConstants.Type;
+import com.ibm.fhir.search.parameters.QueryParameter;
+
+/**
+ * Helper functions to obtain Operator values from QueryParameter objects
+ */
+public class OperatorUtil {
+
+    /**
+     * Get the operator we need to use for matching values for this parameter
+     * @return
+     */
+    public static Operator getOperator(QueryParameter queryParameter) {
+        String operator = LIKE;
+        Modifier modifier = queryParameter.getModifier();
+
+        // In the case where a URI, we need specific behavior/manipulation
+        // so that URI defaults to EQ, unless... BELOW
+        if (Type.URI.equals(queryParameter.getType())) {
+            if (modifier != null && Modifier.BELOW.equals(modifier)) {
+                operator = LIKE;
+            } else {
+                operator = EQ;
+            }
+        } else if (modifier != null) {
+            operator = modifierOperatorMap.get(modifier);
+        }
+
+        if (operator == null) {
+            operator = LIKE;
+        }
+
+        return convert(operator);
+    }
+
+    /**
+     * Convert the operator string value to its enum equivalent
+     * @param op
+     * @return
+     */
+    public static Operator convert(String op) {
+        final Operator result;
+        switch (op) {
+        case LIKE:
+            result = Operator.LIKE;
+            break;
+        case EQ:
+            result = Operator.EQ;
+            break;
+        case NE:
+            result = Operator.NE;
+            break;
+        case LT:
+            result = Operator.LT;
+            break;
+        case LTE:
+            result = Operator.LTE;
+            break;
+        case GT:
+            result = Operator.GT;
+            break;
+        case GTE:
+            result = Operator.GTE;
+            break;
+        default:
+            throw new IllegalArgumentException("Operator not supported: " + op);
+        }
+
+        return result;
+    }
```

Comment:
General pattern we follow is to introduce a from or of in the enum which accomplishes this. I'd recommend pushing this into the enum.

Reply:
This is just an interim workaround until we get rid of the old query stuff. The operator value should've been an enum to begin with and never a series of string constants. I'm definitely not going to pollute the new Operator enum with SQL text values. That breaks separation of concerns. The mapping is the role of the renderer/visitor to translate a given Operation into the correct SQL text string.